### PR TITLE
Cleanup SearchStore

### PR DESCRIPTION
**Pull request by Nokel81**
- fix spelling and grammer in docs
- add appropriate visibility and accessibility modifiers

Signed-off-by: Sebastian Malton <sebastian@malton.name>

This is not an exported type so the fixing of function names is not a breaking change.

### CODE REVIEW

File: src/common/search-store.ts [hunk 1]
```diff
@@ -36,28 +36,28 @@ export class SearchStore {
   }
 
   /**
-   * Does searching within text array, create a list of search keyword occurences.
-   * Each keyword "occurency" is saved as index of the the line where keyword founded
+   * Does searching within text array, create a list of search keyword occurrences.
+   * Each keyword "occurrence" is saved as index of the the line where keyword founded
```
Comment (jim-docker) [inline src/common/search-store.ts]: While you're fixing typos...
```suggestion
   * Each keyword "occurrence" is saved as index of the line where keyword found
```

**Review by jim-docker — DISMISSED** [review]

File: src/common/search-store.ts [hunk 2]
```diff
@@ -19,49 +19,52 @@ export class SearchStore {
...
+    if (this.occurrences[this.activeOverlayIndex] === undefined) {
+      this.activeOverlayIndex = this.getNextOverlay(true);
+    }
   }
 
   /**
-   * Does searching within text array, create a list of search keyword occurences.
-   * Each keyword "occurency" is saved as index of the the line where keyword founded
+   * Does searching within text array, create a list of search keyword occurrences.
+   * Each keyword "occurrence" is saved as index of the the line where keyword was found
```
Comment (jim-docker) [inline src/common/search-store.ts]:
```suggestion
   * Each keyword "occurrence" is saved as index of the line where keyword was found
```

**Review by jim-docker — DISMISSED** [review]

**Review by jim-docker — DISMISSED** [review]

File: src/common/search-store.ts [hunk 3]
```diff
@@ -3,9 +3,34 @@ import { dockStore } from "../renderer/components/dock/dock.store";
...
 
 export class SearchStore {
-  @observable searchQuery = ""; // Text in the search input
-  @observable occurrences: number[] = []; // Array with line numbers, eg [0, 0, 10, 21, 21, 40...]
-  @observable activeOverlayIndex = -1; // Index withing the occurences array. Showing where is activeOverlay currently located
+  /**
+   * An utility methods escaping user string to safely pass it into new Regex(variable)
+   * @param value Unescaped string
+   */
+  public static escapeRegex(value: string): string {
```
Comment (jim-docker) [inline src/common/search-store.ts]: You better change the PR title or do this in a separate PR...

File: src/common/search-store.ts [hunk 4]
```diff
@@ -3,9 +3,34 @@ import { dockStore } from "../renderer/components/dock/dock.store";
...
+   * An utility methods escaping user string to safely pass it into new Regex(variable)
+   * @param value Unescaped string
+   */
+  public static escapeRegex(value?: string): string {
+    return value ? value.replace(/[\-\[\]{}()*+?.,\\\^$|#\s]/g, "\\$&") : "";
+  }
+
+  /**
+   * Text in the search input
+   *
```
Comment (aleksfront) [inline src/common/search-store.ts:16]: We can remove such empty lines, since we're not using this style in the codebase.

File: src/common/search-store.ts [hunk 5]
```diff
@@ -19,49 +44,46 @@ export class SearchStore {
...
-    const occurences: number[] = [];
-
-    text.forEach((line, index) => {
-      const regex = new RegExp(this.escapeRegex(query), "gi");
-      const matches = [...line.matchAll(regex)];
-
-      matches.forEach(() => occurences.push(index));
-    });
+  private findOccurrences(lines?: string[], query?: string): number[] {
+    lines ||= [];
```
Comment (aleksfront) [inline src/common/search-store.ts]: Maybe better use default parameter?
```
private findOccurrences(lines: string[] = [], query?: string): number[] {
```

File: src/common/search-store.ts (hunk 4 above)
Comment (Nokel81) [reply src/common/search-store.ts:16]: Do you mean between the line that begins "Text in ..." and the "@observable"? See `src/main/cluster.ts:239`

File: src/common/search-store.ts (hunk 5 above)
Comment (Nokel81) [reply src/common/search-store.ts]: I tried that but `lines` is sometimes `null`. I will look again to see why.

**Review by aleksfront — APPROVED** [review]
LGTM

**Review by jim-docker — APPROVED** [review]
